umh-core/pkg/service/redpanda/redpanda.go

```diff
@@ -0,0 +1,813 @@
+		}
+		return 0
+	}
+
+	// Helper function to get label value
+	getLabel := func(m *dto.Metric, name string) string {
+		for _, label := range m.Label {
+			if label.GetName() == name {
+				return label.GetValue()
+			}
+		}
+		return ""
+	}
+
+	// Process each metric family
+	for name, family := range mf {
+		switch {
+		// Infrastructure metrics
+		case name == "redpanda_storage_disk_free_bytes":
+			if len(family.Metric) > 0 {
+				metrics.Infrastructure.Storage.FreeBytes = int64(getValue(family.Metric[0]))
+			}
+		case name == "redpanda_storage_disk_total_bytes":
+			if len(family.Metric) > 0 {
+				metrics.Infrastructure.Storage.TotalBytes = int64(getValue(family.Metric[0]))
+			}
+		case name == "redpanda_storage_disk_free_space_alert":
+			if len(family.Metric) > 0 {
+				metrics.Infrastructure.Storage.FreeSpaceAlert = getValue(family.Metric[0]) == 0
+			}
+		case name == "redpanda_uptime_seconds_total":
+			if len(family.Metric) > 0 {
+				metrics.Infrastructure.Uptime.Uptime = int64(getValue(family.Metric[0]))
+			}
+		case name == "redpanda_cluster_topics":
+			if len(family.Metric) > 0 {
+				metrics.Cluster.Topics = int64(getValue(family.Metric[0]))
+			}
+		case name == "redpanda_cluster_unavailable_partitions":
+			if len(family.Metric) > 0 {
+				metrics.Cluster.UnavailableTopics = int64(getValue(family.Metric[0]))
+			}
+		case name == "redpanda_kafka_request_bytes_total":
+			// Based on the label redpanda_request, we can determine if it's bytes in or bytes out
+			// We need to check the label value for that
+			for _, metric := range family.Metric {
+				label := getLabel(metric, "redpanda_request")
+				if label == "produce" {
+					metrics.Throughput.BytesIn = int64(getValue(metric))
+				} else if label == "consume" {
+					metrics.Throughput.BytesOut = int64(getValue(metric))
+				}
+			}
+
+		}
+	}
+
+	return metrics, nil
+}
+
+// GetHealthCheckAndMetrics returns the health check and metrics of a Redpanda service
+func (s *RedpandaService) GetHealthCheckAndMetrics(ctx context.Context, tick uint64) (RedpandaStatus, error) {
+	start := time.Now()
+	defer func() {
+		metrics.ObserveReconcileTime(logger.ComponentRedpandaService, constants.RedpandaServiceName, time.Since(start))
+	}()
+
+	if ctx.Err() != nil {
+		return RedpandaStatus{}, ctx.Err()
+	}
+
+	// Skip health checks and metrics if the service doesn't exist yet
+	// This avoids unnecessary errors in Status() when the service is still being created
+	if _, exists := s.s6Manager.GetInstance(constants.RedpandaServiceName); !exists {
+		return RedpandaStatus{
+			HealthCheck: HealthCheck{
+				IsLive:  false,
+				IsReady: false,
+			},
+			Metrics: Metrics{},
+			Logs:    []s6service.LogEntry{},
+		}, nil
+	}
+
+	baseURL := "http://localhost:9644/"
+	metricsEndpoint := "public_metrics"
+	livenessEndpoint := "ping"
+	readinessEndpoint := "ready"
+
+	// Helper function to make HTTP requests with context
+	doRequest := func(endpoint string) (*http.Response, error) {
```

Comment:
maybe double check my latest code here, I think I also added a timeout context of 1 ms to it after observing what is normnal

Reply:
Couldnt find any